target/linux/realtek/files-5.15/drivers/net/dsa/rtl83xx/rtl930x.c

```diff
@@ -2431,6 +2431,14 @@ static void rtl930x_led_init(struct rtl838x_switch_priv *priv)
 	/* Set LED mode to serial (0x1) */
 	sw_w32_mask(0x3, 0x1, RTL930X_LED_GLB_CTRL);
 
+	// Set LED active state
+	if (of_property_read_bool(node, "active-high")) {
+		// LED_ACTIVE
+		sw_w32_mask(0x00, 1 << 22, RTL930X_LED_GLB_CTRL);
```

Comment:
Regardless of how the inversion will be signaled to the driver, are you sure this works as intended? I.e. does omitting `active-high` result in inverted LED behaviour?

I'm asking because bit 22 (you really should be using a `#define` for that) is [called](https://svanheule.net/realtek/mango/register/led_glb_ctrl) `LEDIC_PAGE_ACCESS`. The bit called `LED_ACTIVE` is actually bit 21, so the comment doesn't make much sense.

The bit actually called `LED_SIGNAL_INVERT` is bit 20, which sounds like it would be the bit you wanted to control.

Reply:
As far as I could see this was working. I checked with the branch I was working on and the same bit was used there. But yeah, this looks like it sets the wrong bit. I'll double-check what happens if I don't set active-high.

Reply:
Actually, it is bit 22 for RTL930x, it is 20 for RTL838X. So the code is correct.

Reply:
Sorry for the noise, I was reading the RTL931x register apparently. So that puts LED_ACTIVE at 22 like you have in the code. I'll try to test this on my XGS1250-12 this week, so I can maybe see what the difference is between LED_ACTIVE and SIGNAL_INVERT.

Reply:
OK, gave it a quick test and it works. Nice :)

Documentation/bindings/leds doesn't have a precedent for `active-high` properties, but `active-low` is in use by some bindings. Could you switch to that and add a `#define RTL930X_LED_GLB_ACTIVE_LOW BIT(22)` (or similar)?

Reply:
Sure, I'll do that.